core/hydromet_stratified.py

```diff
@@ -557,4 +748,119 @@ def plot_max_potential_retention_cond_runoff(GEV_parameters_E: np.ndarray, PMP:
     ax.set_ylabel('$p_S(S | Q)$')
     ax.set_title('Conditional Max Potential Retention Distribution')
     plt.tight_layout()
-    return None
\ No newline at end of file
+    return None
+
+def precip_to_runoff_h1(hydro_events:np.ndarray,nrcs_precip_table_dir: pl.WindowsPath,
+                     precip_data: pd.DataFrame, df_weights_rainfall: pd.DataFrame, CN: int, display_print = False) -> pd.DataFrame:
+    """Takes the events, precipitation data, nrcs temporal distribution, CN and applies the CN reduction method to
+    obtain a runoff curve for each recurrence interval
+    """
+    runoff_distros1 = {}
+    prep_curves = pd.DataFrame(columns = hydro_events.astype(float))
+    for evnt in hydro_events:
+        dist_df = get_hyeto_input_data_nrcs(nrcs_precip_table_dir, evnt, display_print)
+        dist_df['precip'] = dist_df['ratio']*precip_data['Median'].loc[evnt]
+        s = S_24hr(CN)
+        ia = IA_24hr(s)
+        runoff_distros1[evnt] = excess_precip(dist_df,ia, s)
+        prep_curves[evnt] = dist_df['hyeto_input']
+    return prep_curves
+
+
+
+def precip_to_runoff_h2(hydro_events:np.ndarray ,nrcs_precip_table_dir: pl.WindowsPath,
+                     precip_data: pd.DataFrame,df_weights_rainfall: pd.DataFrame, CN: int, display_print = False) -> pd.DataFrame:
+    """Takes the events, precipitation data, nrcs temporal distribution, CN and applies the CN reduction method to
+    obtain a runoff curve for each recurrence interval. Also applies to events beyond published Atlas 14 values. 
+    """
+    runoff_distros1 = {}
+    prep_curves = pd.DataFrame(columns = hydro_events)
+    hyeto_graphs = np.where(df_weights_rainfall.index.to_numpy().astype(int)<1000, df_weights_rainfall.index.to_numpy().astype(int),  1000).tolist()
+    for event, hyetograph in zip(hydro_events, hyeto_graphs):
+        dist_df = get_hyeto_input_data_nrcs(nrcs_precip_table_dir, hyetograph, display_print)
+        dist_df['precip'] = dist_df['ratio']*df_weights_rainfall['P_Mean_in'].loc[event]
+        s = S_24hr(CN)
+        ia = IA_24hr(s)
+        runoff_distros1[event] = excess_precip(dist_df,ia, s)
+        prep_curves[event] = dist_df['hyeto_input']
+    return prep_curves
+    
+def precip_to_runoff_h3(hydro_events:np.ndarray ,nrcs_precip_table_dir: pl.WindowsPath,
+                     precip_data: pd.DataFrame,df_weights_rainfall: pd.DataFrame, display_print = False) -> pd.DataFrame:
+    """Takes the events, precipitation data, nrcs temporal distribution, selected CNs and applies the CN 
+    reduction method to obtain a runoff curve for each recurrence interval. 
+    """  
+    hyeto_graphs = np.where(hydro_events <1000, hydro_events, 1000).astype(int).tolist()
+    runoff_distros1 = {}
+    hydro_events = df_weights_rainfall.index.to_numpy().tolist()
```

Comment:
No need to convert an index object to a numpy array and then to a list. Do: list(df_weights_rainfall.index)

Reply:
Thanks. Change made.